addons/mrp_subcontracting/models/stock_move.py

```diff
@@ -239,3 +239,9 @@ def _update_subcontract_order_qty(self, new_quantity):
                         'mo_id': production.id,
                         'product_qty': production.product_uom_qty - quantity_to_remove
                     }).change_prod_qty()
+
+    def _is_subcontract_return(self):
+        self.ensure_one()
+        origin_is_subcontract = self.move_orig_ids and all(self.move_orig_ids.mapped('is_subcontract'))
```

Comment:
I would use returned_move_id and origin_returned_move_id keys to get the connection between the subcontract move and the return move.

Reply:
Nice, didn't know it existed